datacite-update-doi-handler/src/main/java/no/unit/nva/datacite/handlers/UpdateDoiEventHandler.java

```diff
@@ -90,12 +90,12 @@ private void makeDoiRegisteredIfPublicationGone(
                     input.getPublicationId(),
                     input.getCustomerId());
 
-        if (e.getStatus().getStatusCode() == GONE.getStatusCode()) {
+        if (e.getStatus() == GONE) {
             logger.info(SHOULD_REMOVE_METADATA_LOG_MESSAGE, input.getPublicationId());
 
             var resource = getMetadata(input, doi);
 
-            if (input.getDuplicateOf() != null) {
+            if (input.getDuplicateOf().isPresent()) {
                 addDuplicateIdentifier(resource, input.getDuplicateOf().toString());
             }
```

Comment:
```
            input.getDuplicateOf().ifPresent(duplicate -> addDuplicateIdentifier(resource, duplicate.toString()));
```
Do you test that the metadata received contains the expected duplicate of relation and value?

Reply:
probably not, let me add a test for that :-)